Likvido.Kubesec/Likvido.Kubesec/KubeCtl.cs

```diff
@@ -31,29 +32,60 @@ public IReadOnlyList<Secret> GetSecrets(string secretsName)
             return secrets;
         }
 
-        public Dictionary<string, IReadOnlyList<Secret>> GetAllSecrets()
+        public List<string> GetExistingNamespaces()
         {
-            var allSecretsDictionary = new Dictionary<string, IReadOnlyList<Secret>>();
-            var result = ExecuteCommand($"get secrets -o json");
+            var allNamespaces = ExecuteCommand($"get namespaces -o custom-columns=:metadata.name");
+            var existingNamespaces = allNamespaces.Split(new string[] { "\n" }, StringSplitOptions.None);
+            return existingNamespaces.Where(n => !string.IsNullOrEmpty(n)).ToList();
+        }
 
-            dynamic deserialized = JsonConvert.DeserializeObject(result);
+        public Dictionary<(string Namespace, string Name), IReadOnlyList<Secret>> GetNamespacesWithSecrets(string @namespace, string namespaceContains, string namespaceRegex)
+        {
+            var allSecretsDictionary = new Dictionary<(string, string), IReadOnlyList<Secret>>();
+            string pattern = null;
 
-            foreach (var item in deserialized.items)
+            if (!string.IsNullOrEmpty(@namespace))
             {
-                if (item.type != "Opaque")
-                {
-                    Console.WriteLine($"Skipping secret '{item.metadata.name}', because it is not Opaque type, but '{item.type}'");
-                    continue;
-                }
+                pattern = $"^{@namespace}$";
+            }
+            else if (!string.IsNullOrEmpty(namespaceContains))
+            {
+                pattern = namespaceContains;
+            }
+            else if (!string.IsNullOrEmpty(namespaceRegex))
+            {
+                pattern = namespaceRegex;
+            }
+            else
+            {
+                pattern = "default";
```

Comment:
also, not a good pattern.
I gave this a second thought.
I'd not convert everything to regex.
It's better to define filter function like `Func<string, bool> filter = null` and define it in each `if`